cmake/FindDPCPP.cmake

```diff
@@ -39,31 +39,35 @@ find_library(DPCPP_LIB_DIR NAMES sycl sycl6 PATHS "${DPCPP_BIN_DIR}/../lib")
 add_library(DPCPP::DPCPP INTERFACE IMPORTED)
 
 set(DPCPP_FLAGS "-fsycl;")
+set(DPCPP_COMPILE_ONLY_FLAGS "")
+
 if(NOT "${DPCPP_SYCL_TARGET}" STREQUAL "")
   list(APPEND DPCPP_FLAGS "-fsycl-targets=${DPCPP_SYCL_TARGET};")
 endif()
-list(APPEND DPCPP_FLAGS "${DPCPP_USER_FLAGS};")
+
+if(NOT "${DPCPP_USER_FLAGS}" STREQUAL "")
+  list(APPEND DPCPP_FLAGS "${DPCPP_USER_FLAGS};")
+endif()
 
 if(NOT "${DPCPP_SYCL_ARCH}" STREQUAL "")
   if("${DPCPP_SYCL_TARGET}" STREQUAL "nvptx64-nvidia-cuda")
     list(APPEND DPCPP_FLAGS "-Xsycl-target-backend")
     list(APPEND DPCPP_FLAGS "--cuda-gpu-arch=${DPCPP_SYCL_ARCH}")
+    list(APPEND DPCPP_COMPILE_ONLY_FLAGS; "-mllvm;-enable-global-offset=false;")
```

Comment:
[QUESTION] why cannot we append this to DPCPP_FLAGS, and need to create the new list `DPCPP_COMPILE_FLAGS`

Reply:
It's in the description, resolving

Reply:
That's to avoid passing `-mllvm -enable-global-offset=false` to the linker in

```cmake
    INTERFACE_COMPILE_OPTIONS "${DPCPP_FLAGS};${DPCPP_COMPILE_ONLY_FLAGS}"
    INTERFACE_LINK_OPTIONS "${DPCPP_FLAGS}"
```